modules/mono/glue/GodotSharp/GodotSharp/Core/Hasher.cs

```diff
@@ -0,0 +1,32 @@
+namespace Godot
+{
+    internal static class Hasher
+    {
+        public static Hash<T1, T2>(T1 item1, T2 item2)
+        {
+            int hash = 0;
+            hash = (hash + item1.GetHashCode() + 17) * 23;
```

Comment:
Where are you sourcing these values? The constants used for hashing are pretty important for correctness

Reply:
I'd suggest looking at `core/templates
/hashfuncs.h`

Reply:
The exact values of `17` and `23` are not important. If two Vector2I's have equal X and Y, their hashes will be equal, which is correctness guarantee. `23` is a multiplier that eliminates hash conflicts in case X == Y. `17` is an offset that elimiates conflict of `new Vector2I (0, 5)` vs `new Vector2I (5, 0)`. The only requirement for `17` and `23` is that they are large and preferably odd. I chose them to be middle range prime numbers.

Reply:
The specific constants affect the collisions of hashes which does matter

Reply:
So I've checked what others do here. I dropped `17`. Any odd prime number can be used in place of miltiplier `23`. Sometimes they pick `31` because `<< 5` computes it quickly.

As for conflict frequency, the minimal conflict occurs between vectors: (1, 0) and (0, 31), which could appear in games on very large boards. The next prime number with similar quick computation trick is 127.